BUILD.md

```diff
@@ -1,482 +1,60 @@
-
-   Repeat this for multiple training runs, each time saving the output to a new
-   `-o samples-N.prof` file.  Ideally you want to exercise all code paths in
-   dosbox-staging (core types, video cards, video modes, sound cards, and audio
-   codecs).
-
-4. Convert your sample profiles into compiler-specific records using tools
-   provided in the `autofdo` package:
-
-   For GCC, run:
-   - `create_gcov --binary=/path/to/fdo-trained/dosbox
-     --profile=samples-1.prof -gcov=samples-1.afdo -gcov_version=1`
-
-     ... for each `.prof` file, creating a corresponding `.afdo` file.
-
-   - At this point, you now have an `.afdo` file for each `.prof` file. Merge
-     the `.afdo`s into a single `curren.afdo` file with:
-
-     `profile_merger -gcov_version=1 -output_file=current.afdo *.afdo`
-
-   For Clang, run:
-
-   - `create_llvm_prof --binary=/path/to/fdo-trained/dosbox
-     --profile=samples-1.prof --out=samples-1.profraw`
-
-     ... for each `*.prof` file, creating a corresponding `.profraw` file.
-
-   - At this point, you now have a `.profraw` file for each `.prof` file. Merge
-     them into a single `current.profraw` file with:
-
-     `llvm-profdata-<version> merge -sample -output=current.profraw *.profraw`
-
-### Build Using AutoFDO Data
-
-You can now use your merged `.afdo` or `.profraw` file to build with the `-m
-fdo` modifier by  placing your `current.afdo/.profraw` file in the repo's root
-directory, or point to it using the FDO_FILE environment variable, and launch
-the build with `./scripts/build.sh -c <gcc or clang> -t release -m lto -m fdo`.
+[Windows]: docs/build-windows.md
+[macOS]: docs/build-macos.md
+[Haiku]: docs/build-haiku.md
 
 ## Meson build snippets
 
 ### Make a debug build
 
-Install dependencies listed in README.md.
+Install dependencies listed in [README.md](README.md). Installing `ccache` is
+optional, but highly recommended - Meson will use it to speed up the build
+process.
 
-Build steps :
+Build steps:
 
 ``` shell
 meson setup build
+ninja -C build
+```
+Directory `build` will contain all compiled files.
 
-# for meson >= 0.54.0
-meson compile -C build
+### Other build types
 
-# for older versions
-ninja -C build
+Meson supports several build types, appropriate for various situations:
+`release` for creating optimized release binaries, `debug` (default) for
+for development or `plain` for packaging.
+
+``` shell
+meson setup -Dbuildtype=release build
 ```
+Detailed documentation: [Meson: Core options][meson-core]
+
+[meson-core]: https://mesonbuild.com/Builtin-options.html#core-options
 
 ### Disabling unwanted dependencies
 
 The majority of dependencies are optional and can be disabled during build.
 
-For example, to compile without SDL2\_net and OpenGL dependencies try:
+For example, to compile without OpenGL dependency try:
 
 ``` shell
-meson setup -Duse_sdl2_net=false -Duse_opengl=false build
+meson setup -Duse_opengl=false build
 ninja -C build
 ```
 
```

Comment:
Regarding test coverage, suggest one last command to help the user open the report in Firefox (or their browser)

Open the report with your browser, for example: `firefox build/meson-logs/coveragereport/index.html`